src/communication/hpxreceiver.h

```diff
@@ -37,9 +42,17 @@ class HPXReceiver : public hpx::components::simple_component_base<HPXReceiver<CA
 
     static hpx::future<boost::shared_ptr<HPXReceiver> > make(const std::string& name, std::size_t rank = 0)
     {
```

Comment:
On 02/16/2016 01:33 PM, Hartmut Kaiser wrote:
> In src/communication/hpxreceiver.h
> <https://github.com/STEllAR-GROUP/libgeodecomp/pull/4#discussion_r53004308>:
> 
>> @@ -37,9 +42,17 @@ class HPXReceiver : public hpx::components::simple_component_base<HPXReceiver<CA
>>  
>>      static hpx::future<boost::shared_ptr<HPXReceiver> > make(const std::string& name, std::size_t rank = 0)
>>      {
> 
> Conceptual question: why do you create a component if all you are
> storing is the |shared_ptr<>| referencing it? Why not directly creating
> the object in this case?

The component is needed here since we need to call an action on the
receiver. The shared pointer is merely an optimization
to get direct access to the component in case you need it without
invoking an action on the GID.

> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/STEllAR-GROUP/libgeodecomp/pull/4/files#r53004308>.
> 


-- 
Thomas Heller
Friedrich-Alexander-Universität Erlangen-Nürnberg
Department Informatik - Lehrstuhl Rechnerarchitektur
Martensstr. 3
91058 Erlangen
Tel.: 09131/85-27018
Fax:  09131/85-27912
Email: thomas.heller@cs.fau.de